File: forge/src/main/scala/com/kotori316/fluidtank/forge/tank/ItemBlockTankForge.scala
package com.kotori316.fluidtank.forge.tank

import java.util.function.Consumer

import com.kotori316.fluidtank.forge.render.RenderItemTank
import com.kotori316.fluidtank.tank.ItemBlockTank
import net.minecraft.client.renderer.BlockEntityWithoutLevelRenderer
import net.minecraftforge.client.extensions.common.IClientItemExtensions

class ItemBlockTankForge(b: BlockTankForge) extends ItemBlockTank(b) {

  override def initializeClient(consumer: Consumer[IClientItemExtensions]): Unit = {
    consumer.accept(new IClientItemExtensions {
      override def getCustomRenderer: BlockEntityWithoutLevelRenderer = RenderItemTank.INSTANCE
    })
  }

}

File: forge/src/main/scala/com/kotori316/fluidtank/forge/tank/ItemBlockTankForge.java
package com.kotori316.fluidtank.forge.tank;

import com.kotori316.fluidtank.forge.render.RenderItemTank;
import com.kotori316.fluidtank.tank.BlockTank;
import com.kotori316.fluidtank.tank.ItemBlockTank;
import net.minecraft.client.renderer.BlockEntityWithoutLevelRenderer;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.world.item.ItemStack;
import net.minecraftforge.client.extensions.common.IClientItemExtensions;
import net.minecraftforge.common.capabilities.ICapabilityProvider;
import org.jetbrains.annotations.Nullable;

import java.util.function.Consumer;

public final class ItemBlockTankForge extends ItemBlockTank {
    public ItemBlockTankForge(BlockTank b) {
        super(b);
    }

    @Override
    public void initializeClient(Consumer<IClientItemExtensions> consumer) {
        super.initializeClient(consumer);
        consumer.accept(new IClientItemExtensions() {
            @Override
            public BlockEntityWithoutLevelRenderer getCustomRenderer() {
                return RenderItemTank.INSTANCE();
            }
        });
    }

    @Override
    public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt) {
        return new TankFluidItemHandler(blockTank().tier(), stack);
    }
}
